api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -368,7 +368,8 @@ private void addFieldValuesFromBigQueryToPreviewList(
               valuePreviewList
                   .get(columnNumber)
                   .addQueryValueItem(
-                      Optional.ofNullable(fieldValueList.get(columnNumber).getValue().toString())
+                      Optional.ofNullable(fieldValueList.get(columnNumber).getValue())
```

Comment:
May be my lack of familiarity with Optionals but this reads off a bit weirdly to me.

Is it applying the map function to the non null value from Optional? If not -> ""?

In my head, I think what makes more sense to me is using Optional to get the non null value and then applying `toString` to it.

Reply:
That's basically what it does. The `Optional.map(Object::toString)` turns an Optional<Object> to an Optional<String>. If the first one is empty, then the second one stays empty, and the `orElse` provides the default value of an empty string.

The most confusing thing is that `getValue()` is the thing that can be null.

It might read better if you declare `final Object value = fieldValueList.get(columnNumber).getValue();` and then pack that into the `ofNullable`.

Overall, I'd say Optional.map() is one of the most powerful and functional things you get from Optional.